File: Java/Desktop/ThirdTask/src/by/bntu/fitr/povt/cng/model/logic/Sort.java
package by.bntu.fitr.povt.cng.model.logic;


import by.bntu.fitr.povt.cng.model.entity.abstracts.Track;
import by.bntu.fitr.povt.cng.model.entity.container.Disk;

public class Sort {


    public static void insert(Disk disk) {
        Track temp;
        int j;
        for(int i = 0; i < disk.getSize() - 1; i++){
            if (disk.getTrackLength(i) > disk.getTrackLength(i + 1)) {
                temp = disk.getTrack(i + 1);
                disk.changeTrack(disk.getTrack(i),i + 1);
                j = i;
                while (j > 0 && temp.getLength() < disk.getTrackLength(j - 1)) {
                    disk.changeTrack(disk.getTrack(j - 1),j);
                    j--;
                }
                disk.changeTrack(temp,j);
            }
        }
    }

//
//
//    public static void bubble(Tree tree) {
//        boolean sorted = true;
//        for (int i = tree.getSize(); i > 0; i--) {
//            for (int j = 0; j < i - 1; j++) {
//                if (tree.getToy(j).getPrice() > tree.getToy(j + 1).getPrice()) {
//                    Toy tmp = tree.getToy(j);
//                    tree.changeToy(tree.getToy(j + 1), j);
//                    tree.changeToy(tmp, j + 1);
//                    sorted = false;
//                }
//                if (sorted){
//                    break;
//                }
//
//            }
//        }
//    }



}

File: Ruby/Object-oriented-programming/Lab/Model/Logic/Sort.rb
require_relative "../../../Lab/Model/Entity/Abstracts/Toy"
require_relative "../../../Lab/Model/Entity/Container/Tree"

class Sort

  def Sort.choice(tree)
    i = 0
    while i < tree.get_size - 1
      if tree.get_price(i) > tree.get_price(i + 1)
        temp = tree.get_toy(i+1)
        tree.change_toy(tree.get_toy(i),i + 1)
        j = i
        while j > 0 and temp.price < tree.get_price(j - 1)
          tree.change_toy(tree.get_toy(j-1), j)
          j -= 1
        end
        tree.change_toy(temp,j)
      end
      i += 1
    end
  end




  def Sort.bubble(tree)
    sorted = true
    i = tree.get_size
    j = 0
    while i > 0
      while j < i - 1
        if tree.get_price(j) > tree.get_price(j + 1)
          tmp = tree.get_toy(j)
          tree.change_toy(tree.get_toy(j + 1),j)
          tree.change_toy(tmp, j + 1)
          sorted = false
        end
        if sorted
          break
        end
        j += 1
      end
      i -=1
    end
  end


end

